symengine/univariate_int.h

```diff
@@ -0,0 +1,82 @@
+/**
+ *  \file polynomial_int.h
+ *  Class for Univariate Int Polynomial Base
+ **/
+#ifndef SYMENGINE_UNIVARIATE_INT_H
+#define SYMENGINE_UNIVARIATE_INT_H
+
+#include <symengine/basic.h>
+
+namespace SymEngine
+{
+
+template <typename Coeff, typename Container, typename Poly>
+class UnivariateIntBase : public Basic
+{
+protected:
+    unsigned int degree_;
+    RCP<const Symbol> var_;
+    Container container_;
```

Comment:
Then `map_int_Expr` is not the right fit here. Maybe something like `UnivariateExprPolynomial` would do. (Not the implementation we have right now, but the one @irislq has started on)